workshop/content/lab1.6_operator.md

```diff
@@ -0,0 +1,179 @@
+        # image that will be deployed
+        - image: quay.io/jkeam/hello-python:latest
+```
+
+Let's verify that our application was deployed.
+
+```
+oc get ksvc
+```
+
+We can also see the public route (URL).
+
+```
+NAME           URL                                                                                      LATESTCREATED     LATESTREADY       READY   REASON
+hello-python   http://hello-python-user-1.apps.cluster-tysons-5b24.tysons-5b24.sandbox891.opentlc.com   hello-python-v1   hello-python-v1   True
+```
+
+Open the URL in your browser and you should see
+
+```
+Hello World!
+```
+
+## Deployment v2
+Let's imagine now that the development team has delivered a new version of the `hello-python` application.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/deployment/hello-python-v2.yml | oc apply -f -
+```
+
+Here's that new deployment file, with comments added.
+
+```
+apiVersion: serving.knative.dev/v1
+kind: Service
+metadata:
+  name: hello-python
+spec:
+  template:
+    metadata:
+      # new name of our revision
+      name: hello-python-v2
+    spec:
+      containers:
+      containers:
+        - image: quay.io/jkeam/hello-python
+          # new environment variable applied to this configuration so we can tell the difference between our service versions
+          env:
+            - name: TARGET
+              value: "World v2"
+```
+
+Now refresh the browser and you should see `Hello World v2`.
+
+## Blue-Green Deployment
+
+Deploying a new Knative service actually creates a new immutable revision and routes all traffic to it.  These immutable revisions allows us to have sophisticated deployment strategies.  The first we will examine is blue-green.
+
+Let's reset all traffic back to v1.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/deployment/hello-python-blue.yml | oc apply -f -
+```
+
+Here's the relevant section from that yaml file that controls the traffic distribution.
+
+```
+  traffic:
+    - tag: v1
+      revisionName: hello-python-v1
+      percent: 100
+    - tag: v2
+      revisionName: hello-python-v2
+      percent: 0
+
+```
+
+This directs all traffic back to v1 of our application.  Refresh the page and you will see the original `Hello World!`.
+
+Then after testing, we get approval to switch to v2, we can apply the following.
+
+```
+curl -sS https://raw.githubusercontent.com/RedHatGov/serverless-workshop-code/deployment/hello-python-green.yml | oc apply -f -
+```
+
+You can see that all traffic has switched over to v2
+
+```
+  traffic:
+    - tag: v1
+      revisionName: hello-python-v1
+      percent: 0
+    - tag: v2
+      revisionName: hello-python-v2
+      percent: 100
+
+```
+
+Refresh the page and you will see the version 2 text.
+
+## Canary Deployment
+
+The immutable revisions coupled with our ability to configure traffic distribution means we can also do Canary deployment.  Let's try that now.
+
```

Comment:
Similar to comment above - can we have something like: If you aren't familiar with the concept of a canary deployment you can check out this link: https://martinfowler.com/bliki/CanaryRelease.html

Reply:
Great idea, will add.

Reply:
Just added.